homeassistant/components/media_player/horizon.py

```diff
@@ -0,0 +1,310 @@
+"""
+Support for the Unitymedia Horizon HD Recorder.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/media_player.horizon/
+"""
+
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    MediaPlayerDevice, PLATFORM_SCHEMA, MEDIA_TYPE_CHANNEL,
+    SUPPORT_NEXT_TRACK, SUPPORT_TURN_ON, SUPPORT_TURN_OFF, SUPPORT_PAUSE,
+    SUPPORT_PLAY, SUPPORT_PLAY_MEDIA, SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_SELECT_SOURCE)
+from homeassistant.const import (CONF_HOST, CONF_NAME, CONF_PORT, STATE_OFF,
+                                 STATE_PAUSED, STATE_PLAYING)
+import homeassistant.helpers.config_validation as cv
+import homeassistant.util as util
+
+REQUIREMENTS = ['einder==0.3.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "Horizon"
+DEFAULT_PORT = 5900
+
+MIN_TIME_BETWEEN_FORCED_SCANS = timedelta(seconds=1)
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
+
+SUPPORT_UM_HORIZON = SUPPORT_NEXT_TRACK | SUPPORT_PAUSE | SUPPORT_PLAY | \
+    SUPPORT_PLAY_MEDIA | SUPPORT_PREVIOUS_TRACK | SUPPORT_SELECT_SOURCE | \
+    SUPPORT_TURN_ON | SUPPORT_TURN_OFF
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Horizon platform."""
+    from einder import Client, keys
+    from einder.exceptions import AuthenticationError
+
+    host = config.get(CONF_HOST)
+    name = config.get(CONF_NAME)
+    port = config.get(CONF_PORT)
+
+    try:
+        client = Client(host, port=port)
+    except (AuthenticationError, OSError) as msg:
+        _LOGGER.error("Connection to %s at %s failed: %s", name, host, msg)
+        return False
```

Comment:
You could raise a PlatformNotReady exception if there is a chance (`OSError`?) for recovery. The AuthenticationError should return `False`.

Reply:
Nothing is checking this return value. Either just `return` or raise `PlatformNotReady` as recommended above.